protocol/auditlog.go

```diff
@@ -11,10 +11,21 @@ import (
 )
 
 type directoryHistory struct {
+	*AudState
 	addr      string
-	signKey   sign.PublicKey
-	snapshots map[uint64]*DirSTR
-	latestSTR *DirSTR
+	snapshots []*DirSTR
```

Comment:
Can we keep using the map as before? Is there any advantages over using a map? 
I'm afraid that we may accidentally insert 2 same DirSTR into snapshots if we use slice.